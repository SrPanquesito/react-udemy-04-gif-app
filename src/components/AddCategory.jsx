import { useState } from "react"
import { PropTypes } from 'prop-types';

export const AddCategory = ({ emitAddCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;

        // onSetCategories( (categories) => [ inputValue, ...categories] );
        emitAddCategory(inputValue.trim())
        setInputValue('');
    }

    return (
        <>
            <form onSubmit={ onSubmit } aria-label="form-test" >
                <input
                    type="text"
                    placeholder="Buscar gifs"
                    value={ inputValue }
                    onChange={ onInputChange }
                />
            </form>
        </>
    )
}

AddCategory.propTypes = {
    emitAddCategory: PropTypes.func.isRequired
}