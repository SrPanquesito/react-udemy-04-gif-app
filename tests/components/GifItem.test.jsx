import { render, screen } from '@testing-library/react';
import { GifItem } from './../../src/components/GifItem';

describe('Pruebas <GifItem />', () => {
    const title = 'Dragon Ball';
    const url = 'https://google.com';

    test('Debe hacer match en el snapchot', () => {
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test(`Debe de mostrar la imagen con el URL y el ALT indicado`, () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url + '/');
        expect(alt).toBe(title);
    })

    test(`Debe mostrar el texto indicado`, () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
});