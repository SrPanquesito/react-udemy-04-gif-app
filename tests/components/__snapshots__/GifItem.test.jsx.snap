// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifItem /> Debe hacer match en el snapchot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://google.com"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
